java-2016/homework-9/IOhomework/src/AthleteEvaluation.java

```diff
@@ -0,0 +1,106 @@
+import java.util.*;
+
+
+/**
+ * Created by Turian Ovidiu.
+ * This class is responsible for calculation the ranking.
+ */
+public class AthleteEvaluation {
+
+
+    public AthleteEvaluation() {
+    }
+
+
+    /**
+     * This method takes a list of Athletes and will return an list sorted by ski time result after calculation the shots.
+     * @param athletes the list of Athletes.
+     * @return list of Athletes sorted by ski time result after calculation the shots.
+     */
+    public List<Athlete> rankingAthletes(List<Athlete> athletes) {
+        calculateShots(athletes);
+        evaluateRanking(athletes);
+        return athletes;
+    }
+
+    /**
+     * This method calculate the shots and will add time to the ski time result if it is necessary.
+     * @param athletes list of Athletes.
+     */
+    private void calculateShots(List<Athlete> athletes) {
+
+        Calendar calendar = Calendar.getInstance();
+
+        for (Athlete athlete : athletes) {
+            calendar.setTime(athlete.getSkiTimeResult());
+
+            for (char c : athlete.getFirstShootingRange().toCharArray()) {
+                if (c == 'o') {
+                    calendar.add(Calendar.SECOND, 10);
+                    athlete.setSkiTimeResult(calendar.getTime());
```

Comment:
no need to set each time. This could have been done outside of the for loop. You update the athlete object each time but you only need to do so after all the penalty was computed.

 for (char c : athlete.getFirstShootingRange().toCharArray()) {
                if (c == 'o') {
                    calendar.add(Calendar.SECOND, 10);
                }
            }
            
            athlete.setSkiTimeResult(calendar.getTime());